test/modern_bpf/test_suites/syscall_enter_suite/access_e.cpp

```diff
@@ -0,0 +1,40 @@
+#include "../../event_class/event_class.h"
+
+#ifdef __NR_access
+
+TEST(SyscallEnter, accessE)
+{
+	auto evt_test = get_syscall_event_test(__NR_access, ENTER_EVENT);
+
+	evt_test->enable_capture();
+
+	/*=============================== TRIGGER SYSCALL  ===========================*/
+
+	int32_t _mode = W_OK;
+	const char *_pathname = "./mock_file";
+	assert_syscall_state(SYSCALL_FAILURE, "access", syscall(__NR_access, _pathname, _mode));
+
+	/*=============================== TRIGGER SYSCALL ===========================*/
+
+	evt_test->disable_capture();
+
+	evt_test->assert_event_presence();
+
+	if(HasFatalFailure())
+	{
+		return;
+	}
+
+	evt_test->parse_event();
+
+	//evt_test->assert_header();
+
+	/*=============================== ASSERT PARAMETERS  ===========================*/
+
+	// Here we have no parameters to assert.
```

Comment:
The enter event pushes the mode event parameter. So this needs to be asserted here.

Reply:
Still getting CI failures on the expected event not being in the buffer for the unit tests.

Reply:
The test cases looks good so far... I think the missing piece here is they are likely not attached due to missing definitions in libpman. Could you append the enter and exit event to the `event_prog_names` in [userspace/libpman/src/events_prog_names.h](https://github.com/falcosecurity/libs/blob/c1d075ffda41dcbdeec0a9fee86f288b7b360d19/userspace/libpman/src/events_prog_names.h#L183)

Reply:
yes @hbrueckner is completely right you need to add 2 entries to `events_prog_names` vector https://github.com/falcosecurity/libs/blob/c1d075ffda41dcbdeec0a9fee86f288b7b360d19/userspace/libpman/src/events_prog_names.h#L24

```
   	[PPME_SYSCALL_ACCESS_E] = "access_e",
	[PPME_SYSCALL_ACCESS_X] = "access_x",
```

and after that we should be ready :)

Reply:
Added these, and updated to use the ringbuf API over auxmap. The access_x test is failing:

```
/home/runner/work/libs/libs/test/modern_bpf/event_class/event_class.cpp:427: Failure
Expected equality of these values:
  m_event_header->len
    Which is: 53
  m_event_len
    Which is: 61
'event_len' in the header is not correct.
```

I'm assuming that the auxmap API is required for access_x because one of the parameters being extracted is a string whose length is unknown at compile time?